src/org/exist/collections/Collection.java

```diff
@@ -2305,4 +2303,32 @@ public QNameRangeIndexSpec getIndexByQNameConfiguration(final DBBroker broker, f
         final IndexSpec idxSpec = getIndexConfiguration(broker);
         return (idxSpec == null) ? null : idxSpec.getIndexByQName(qname);
     }
+
+    @Override
+    public ResourceMetadata getMetadata() {
+        return new CollectionMetadata();
+    }
+
+    class CollectionMetadata implements ResourceMetadata {
+
+        @Override
+        public String getMimeType() {
+            return "collection";
+        }
+
+        @Override
+        public long getCreated() {
+            return Collection.this.getCreationTime();
+        }
+
+        @Override
+        public long getLastModified() {
+            return getCreated();
+        }
+    }
+
+    public boolean isFolder() {
```

Comment:
Sorry... what is a `Folder`? Are we introducing a new concept? How is a `Folder` different to a `Collection`?